test/e2e/specs/editor/various/mentions.spec.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'autocomplete mentions', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.createUser( 'testuser', 'Jane', 'Doe' );
+	} );
+
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.deleteUser( 'testuser' );
+	} );
+
+	test( 'should insert mention', async ( { page, editor } ) => {
+		await editor.clickBlockAppender();
+		await page.keyboard.type( 'I am @a' );
+		await page.locator( 'button', { hasText: 'adminadmin' } ).click();
```

Comment:
It'd be good to use keyboard input, the same way the old test worked. So once the popover is visible, hit Enter followed by typing a `.`

You can use the `.waitFor()` method on a locator to wait for the autocomplete option to become visible, and then once its present continue with the same keyboard input as the old test did.

If it's possible to use a role selector for the locator that would also be great. Possibly something like:
```
role=option >> text="admin"
```

Though I've noticed sometimes the role might be button, despite the dev tools showing something different:
```js
role=button >> text="admin"
```

Reply:
Hi @talldan 
Not sure why `'.components-autocomplete__result'` element resolves to two elements while typing `@a` should resolve to one only. Do you have any idea?

Reply:
Hi @juhi123, I see what's happening. It's because 'admin' is repeated twice in the button:
![Screen Shot 2022-08-25 at 1 03 57 pm](https://user-images.githubusercontent.com/677833/186579284-b82fe144-ff6f-43c6-a882-455036e9405e.png)

The element with the text 'admin' on the left is one element, and the element with the text 'admin' on the right is the second element.

I think the best option will be this:
```js
await page.locator( 'role=option', { hasText: 'admin' } )
```

That works because it only selects the button element.

Reply:
Hi @talldan 
I didn't mean about this `await page.locator( 'button', { hasText: 'adminadmin' } ).click();` I meant about the autocomplete popup when using below code
`await page.locator( '.components-autocomplete__result' ).waitFor();
 await page.keyboard.press( 'Enter' );
`

It gives this error 

 ```locator.waitFor: Error: strict mode violation: ".components-autocomplete__result" resolved to 2 elements:
        1) <button type="button" role="option" aria-selected="true"…>…</button> aka playwright.$("button[role="option"]:has-text("adminadmin")")
        2) <button type="button" role="option" aria-selected="false…>…</button> aka playwright.$("text=Jane Doetestuser")```